src/OpenTelemetry.AutoInstrumentation/Instrumentations/MongoDb/MongoClientIntegration.cs

```diff
@@ -65,12 +60,10 @@ public static class MongoClientIntegration
         var existingDelegate = clusterConfiguratorProperty?.GetValue(settings) as Delegate;
 
         clusterConfiguratorProperty?.SetValue(settings, Delegate.Combine(existingDelegate, setListenerDelegate));
-#endif
 
         return CallTargetState.GetDefault();
     }
 
-#if NET6_0_OR_GREATER
     private static object GetInstrumentationOptions()
     {
         Type optionsType = Type.GetType("MongoDB.Driver.Core.Extensions.DiagnosticSources.InstrumentationOptions, MongoDB.Driver.Core.Extensions.DiagnosticSources")!;
```

Comment:
Can take direct reference now

Reply:
It seems better to change to direct reference in a separate PR.

Reply:
It is great comment. This type was removed https://github.com/jbogard/MongoDB.Driver.Core.Extensions.OpenTelemetry/commit/7652f6286d7be1fbd1c50bb37b57966de004d6f1

I will check how it impact us and if we really need to ship this package at all.

Reply:
#2454